core/src/main/java/org/infinispan/persistence/manager/TracedPersistenceManager.java

```diff
@@ -0,0 +1,117 @@
+   }
+
+   @Override
+   public <K, V> CompletionStage<MarshallableEntry<K, V>> loadFromAllStores(Object key, int segment, boolean localInvocation, boolean includeStores) {
+      return decorateCommand("loadFromAllStores", super.loadFromAllStores(key, segment, localInvocation, includeStores));
+   }
+
+   @Override
+   public CompletionStage<Long> approximateSize(Predicate<? super StoreConfiguration> predicate, IntSet segments) {
+      return decorateCommand("approximateSize", super.approximateSize(predicate, segments));
+   }
+
+   @Override
+   public CompletionStage<Long> size(Predicate<? super StoreConfiguration> predicate, IntSet segments) {
+      return decorateCommand("size", super.size(predicate, segments));
+   }
+
+   @Override
+   public CompletionStage<Void> writeToAllNonTxStores(MarshallableEntry marshalledEntry, int segment,
+                                                      Predicate<? super StoreConfiguration> predicate, long flags) {
+      return decorateCommand("writeToAllNonTxStores", super.writeToAllNonTxStores(marshalledEntry, segment, predicate, flags));
+   }
+
+   @Override
+   public CompletionStage<Void> commitAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+                                                  Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("commitAllTxStores", super.commitAllTxStores(txInvocationContext, predicate));
+   }
+
+   @Override
+   public CompletionStage<Void> rollbackAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+                                                    Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("rollbackAllTxStores", super.rollbackAllTxStores(txInvocationContext, predicate));
+   }
+
+   @Override
+   public <K, V> CompletionStage<Void> writeEntries(Iterable<MarshallableEntry<K, V>> iterable,
+                                                    Predicate<? super StoreConfiguration> predicate) {
+      return decorateCommand("writeEntries", super.writeEntries(iterable, predicate));
+   }
+
+   @Override
+   public CompletionStage<Long> performBatch(TxInvocationContext<AbstractCacheTransaction> ctx,
+                                             TriPredicate<? super WriteCommand, Object, MVCCEntry<?, ?>> commandKeyPredicate) {
+      return decorateCommand("performBatch", super.performBatch(ctx, commandKeyPredicate));
+   }
+
+   private InfinispanSpan requestStart(String operationName) {
+      InfinispanSpanAttributes attributes = new InfinispanSpanAttributes.Builder(SpanCategory.PERSISTENCE)
+            .withCache(cache.getName(), configuration)
+            .build();
+
+      return telemetryService.startTraceRequest(operationName, attributes);
+   }
+
+   private <T> CompletionStage<T> decorateCommand(String operationName, CompletionStage<T> operation) {
+      if (super.isEnabled()) {
```

Comment:
```suggestion
      if (super.isEnabled() && configuration.tracing().enabled(SpanCategory.PERSISTENCE)) {
```
Since we want this property to be runtime enablable we have to check it on _every_ invocation.

Reply:
the check `configuration.tracing().enabled(SpanCategory.PERSISTENCE)`  is implemented  here => https://github.com/fax4ever/infinispan/blob/1582088abf5e345320976b8c5f35abc2f9944e52/core/src/main/java/org/infinispan/telemetry/InfinispanSpanAttributes.java#L33

Reply:
Yes, but that is after it has already allocated the InfinispanSpanAttributes and its builder and registering a BiConsumer callback to whenComplete.. we want when persistence is not enabled to not allocate anything on these paths. In the case when persistence is disabled all we want is to check the configuration boolean without allocating additional objects.

Reply:
Right. Going to add the suggestion. Thank you @wburns